### PR TITLE
I give up on this branch... needs to break into more branches before merging

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
as title. 
Now this branch hasn't been connected to backend.
Loading a single college has not been fully supported(though it may work).

### CODE REVIEW

File: src/main/webapp/js/finscholar/commonlistview/commonlistview.js [hunk 1]
```diff
@@ -52,13 +52,14 @@ class CommonListView {
   async init(tableContainer) {
     tableContainer.innerHTML = commonlistview({pageIndex: this.optionIndex_});
     this.container_ = googDom.getElement(ITEM_CONTAINER_ID)
-    window.addEventListener('scroll', await this.bindedScrollHandler_);
+    window.addEventListener('scroll', this.bindedScrollHandler_);
```
Comment (MikolJiron) [inline src/main/webapp/js/finscholar/commonlistview/commonlistview.js]: Was it necessary for this part to be asynchronous?

File: src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.js [hunk 2]
```diff
@@ -41,23 +41,20 @@ class ScholarshipPageView {
    * Render the scholarship page.
    * @public
    * @param {!Element} container - The DOM element for single scholarship page.
+   * @param {string} id - The unique id of the scholarship.
    */
-  async renderScholarship(container) {
-    // In the prototype, the id is set to 0 by default. Later we'll pass in id as parameter.
-    const id = 0;
+  async renderView(container, id) {
```
Comment (MikolJiron) [inline src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.js]: It seems right now it's just rendering the view for scholarships, will this be generalized for the other views?

**Review by MikolJiron — APPROVED** [review]

**Comment by MikolJiron** [comment]
Also, the query methods for batching are ready, so we should probably hook them up ASAP.

File: src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.js (hunk 2 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/finscholar/scholarshippageview/scholarshippageview.js]: I'll make a common parent class for scholarship page view and college page view.

File: src/main/webapp/js/finscholar/commonlistview/commonlistview.js (hunk 1 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/finscholar/commonlistview/commonlistview.js]: Actually no. The handler itself is asynchronous anyways lol.

File: src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js [hunk 3]
```diff
@@ -60,6 +61,10 @@ class HomePageController extends PageController {
                                CollegeListView, 
                                 ScholarshipListView, 
                                ];
+    /** @private @const {!function(!jsaction.ActionFlow): Promise<undefined>} */
+    this.bindedListItemOnlickHandler_ = this.handleListItemOnclickEvent_.bind(this);
+    /** @private {CommonListView} Registers the current list (if any) opened. */
+    this.currentList_ = null;
```
Comment (Lorraine-Lyu) [inline src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js]: @hzimz I thought about changing this to onFocus but later I found my onscoll event manipulates instance variables in listHandlers, which are created after one navbar is clicked...

**Review by MikolJiron — APPROVED** [review]

**Comment by CarlosM10011** [comment]
@Lorraine-Lyu I ran the js compiler linter, and found these errors:

```
src/main/webapp/js/datahandlers/collegelistdatahandler.js:22: WARNING - [JSC_UNUSED_LOCAL_ASSIGNMENT] Value assigned to local variable COLLEGE_LIST_ENDPT is never read
  22| const COLLEGE_LIST_ENDPT = '';
            ^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/collegelistdatahandler.js:47: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  47| exports = {CollegeListDataHandler}
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/collegelistdatahandler.js:32: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  32|   /** @returns The total number of scholarship stored in backend. */
            ^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:33: WARNING - [JSC_STRING_CONTINUATION] String continuations are not recommended. See https://google.github.io/styleguide/jsguide.html#features-strings-no-line-continuations
  33| const ACTION_SERVER_ERROR = 'The database failed to retrieve the college. \
                                  ^

src/main/webapp/js/datahandlers/scholarshipdatahandler.js:40: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Object is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  40|    * @return {Object} - The object map representing a scholarship's data.
                    ^^^^^^

src/main/webapp/js/datahandlers/scholarshipdatahandler.js:58: WARNING - [JSC_UNUSED_LOCAL_ASSIGNMENT] Value assigned to local variable key is never read
  58|     for (key in data) {
               ^^^

src/main/webapp/js/datahandlers/scholarshipdatahandler.js:18: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'SoyMap' is never referenced in this file
  18| const {Map: SoyMap} = goog.require('soy.map');
                  ^^^^^^

src/main/webapp/js/datahandlers/scholarshipdatahandler.js:87: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  87|    * @return The formatted scholarship JS object map.
           ^

src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js:51: WARNING - [JSC_MISSING_RETURN_JSDOC] Function with non-trivial return must have JSDoc indicating the return type.
  51|   async getNextBatch(itemsPerBatch, lastIndex) {
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  52|     let scholarshipList = [];
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
  63|     return {scholarships: scholarshipList};	
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  64|   }
      ^^^

src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js:82: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  82| exports = {ScholarshipListDataHandler}
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js:42: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  42|   /** @returns The total number of scholarship stored in backend. */
            ^

src/main/webapp/js/datahandlers/scholarshiplistdatahandler.js:68: ERROR - [JSC_JSDOC_MISSING_TYPE_WARNING] Missing type declaration.
  68|    * @return The scholarship list item.
           ^



src/main/webapp/js/finscholar/commonlistview/commonlistview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const JsactionActionFlow = goog.require('jsaction.ActionFlow');
const googDom = goog.require('goog.dom');
const {BasicView} = goog.require('basicview');
const {ScholarshipListDataHandler} = goog.require('datahandlers.scholarshiplistdatahandler');
const {collegelistitems, commonlistview, endoflist, loading, scholarshiplistitems} = goog.require('finscholar.commonlistview.templates');

  19| const JsactionActionFlow = goog.require('jsaction.ActionFlow');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:51: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Element is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  51|     /** @private @const {function({scholarships : !Array<?>}):Element} */
                                                                    ^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:63: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] Function is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  63|     /** @private @const {!function():Promise<undefined>} */
                               ^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:63: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Promise is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  63|     /** @private @const {!function():Promise<undefined>} */
                                           ^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:66: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] Function is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  66|     /** @private @const {!function(number):Promise<undefined>} */
                               ^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:66: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Promise is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  66|     /** @private @const {!function(number):Promise<undefined>} */
                                                 ^^^^^^^

src/main/webapp/js/finscholar/commonlistview/commonlistview.js:19: ERROR - [JSC_EXTRA_REQUIRE_WARNING] extra require: 'JsactionActionFlow' is never referenced in this file
  19| const JsactionActionFlow = goog.require('jsaction.ActionFlow');
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/singlepageview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const googSoy = goog.require('goog.soy');
const {BasicView} = goog.require('basicview');
const {ScholarshipDataHandler} = goog.require('datahandlers.scholarshipdatahandler');

  19| const {BasicView} = goog.require('basicview');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/singlepageview.js:37: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] Function is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  37|      * @private @const {!function(?): googSoy.data.SanitizedHtml}
                              ^

src/main/webapp/js/finscholar/singlepageview.js:37: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] googSoy.data.SanitizedHtml is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  37|      * @private @const {!function(?): googSoy.data.SanitizedHtml}
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Pull request closed without merging**